floodlight/io/dfl.py

```diff
@@ -0,0 +1,324 @@
+from pathlib import Path
+from typing import Dict, Tuple, Union
+
+from lxml import etree
+import datetime as dt
+import numpy as np
+
+from floodlight.core.code import Code
+from floodlight.core.pitch import Pitch
+from floodlight.core.xy import XY
+
+
+def _read_pitch_from_mat_info(filepath_mat_info: Union[str, Path]) -> Pitch:
+    """Reads match_information XML file and returns the playing Pitch.
+
+    Parameters
+    ----------
+    filepath_mat_info: str or pathlib.Path
+        Full path to XML File where the Match Information data in DFL format is saved.
+
+    Returns
+    -------
+    pitch: Pitch
+        Pitch object with actual pitch length and width.
+    """
+    # set up XML tree
+    tree = etree.parse(str(filepath_mat_info))
+    root = tree.getroot()
+
+    # parse pitch
+    length = root.find("MatchInformation").find("Environment").get("PitchX")
+    length = float(length) if length else None
+    width = root.find("MatchInformation").find("Environment").get("PitchY")
+    width = float(width) if width else None
+    pitch = Pitch.from_template(
+        "dfl",
+        length=length,
+        width=width,
+    )
```

Comment:
```suggestion
        sport="football"
    )
```

Reply:
I assumed that DFL format always relates to football therefore setting it default in the Pitch object.
 
 It is now modified in the code and in the Pitch object.